winforms/src/toga_winforms/widgets/base.py

```diff
@@ -147,6 +160,11 @@ def refresh(self):
         self.interface.intrinsic.width = at_least(self.interface._MIN_WIDTH)
         self.interface.intrinsic.height = at_least(self.interface._MIN_HEIGHT)
         self.rehint()
+        # Background color needs to be reapplied on widget refresh as WinForms
+        # doesn't actually support transparency. It just copies the parent's
+        # BackColor to the widget. So, if a widget's parent changes then we need
+        # to reapply background_color to copy the new parent's BackColor.
+        self.set_background_color(self.interface.style.background_color)
```

Comment:
I'm not sure where should I move the re-application call to. I had thought refresh was to be used for reapplying changes related to both layout changes and style changes.

Reply:
No - refresh is an internal trigger when something affecting widget layout has occurred - something that could change the *size* of the widget. 

`set_background_color()` will be invoked when any style element that alters the background color of a widget changes. 

What is the sequence of events that requires a call in refresh? How do you manufacture a problem that is resolved by this line of code?

Reply:
The background color needs to be reapplied when the widget's parent changes. Since refresh should be used for layout, I have moved the call to reapply background color, to the widget's `container setter` - as that is the place where the widget's parent changes.